### PR TITLE
Add Rook/Ceph storage

### PR DESCRIPTION
**Pull request by ysksuzuki**
- add manifest to ArgoCD
- add storage test
- add storage test workflow to circleci/config.yaml

Signed-off-by: Yusuke Suzuki <yusuke-suzuki@cybozu.co.jp>

### CODE REVIEW

File: rook/base/ceph-for-ss/cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    storageClassDeviceSets:
+      - name: set1
+        count: 18
+        volumeClaimTemplates:
+          - metadata:
+              name: data
+            spec:
+              resources:
+                requests:
+                  storage: 5Gi
```
Comment (morimoto-cybozu) [inline rook/base/ceph-for-ss/cluster.yaml:36]: This seems too small.  Is this a dynamic parameter?

File: rook/base/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+   - upstream/common.yaml
+   - upstream/operator.yaml
+   - ceph-for-cs/cephblockpool.yaml
+   - ceph-for-cs/cluster.yaml
+   - ceph-for-cs/rbac.yaml
+   - ceph-for-ss/cephobjectstore.yaml
+   - ceph-for-ss/cluster.yaml
+   - ceph-for-ss/rbac.yaml
+patchesStrategicMerge:
+   - operator.yml
```
Comment (morimoto-cybozu) [inline rook/base/kustomization.yaml:13]: "operator.yaml" is preferable.

File: rook/base/ceph-for-cs/cluster.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                requests:
+                  storage: 1Ti
+              # IMPORTANT: Change the storage class depending on your environment (e.g. local-storage, gp2)
+              storageClassName: topolvm-provisioner
+              volumeMode: Block
+              accessModes:
+                - ReadWriteOnce
+        placement:
+          topologySpreadConstraints:
+          - maxSkew: 1
```
[the remark this answers is not in the document]
Comment (ysksuzuki) [reply rook/base/ceph-for-cs/cluster.yaml]: Fixed

File: rook/base/ceph-for-cs/cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
+apiVersion: ceph.rook.io/v1
+kind: CephCluster
+metadata:
+  name: ceph-for-cs
+  namespace: ceph-for-cs
+  annotations:
+    argocd.argoproj.io/sync-wave: "1"
+spec:
+  dataDirHostPath: /var/lib/rook
```
[the remark this answers is not in the document]
Comment (ysksuzuki) [reply rook/base/ceph-for-cs/cluster.yaml:9]: We'll work on it in the following task. Currently, it contains logs and crash files, so losing these files doesn't lead to a serious problem.
https://app.zenhub.com/workspaces/neco-5ca28d055128c041a68c0a8f/issues/cybozu-go/neco/736

File: rook/base/ceph-for-ss/cephobjectstore.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    failureDomain: zone
+    replicated:
+      size: 3
+  preservePoolsOnDelete: true
+  gateway:
+    type: s3
+    sslCertificateRef:
+    port: 80
+    securePort:
+    instances: 1
```
[the remark this answers is not in the document]
Comment (ysksuzuki) [reply rook/base/ceph-for-ss/cephobjectstore.yaml:23]: We'll work on it in the following epic.
https://app.zenhub.com/workspaces/neco-5ca28d055128c041a68c0a8f/issues/cybozu-go/neco/755

File: rook/base/ceph-for-ss/cluster.yaml (hunk 1 above)
Comment (ysksuzuki) [reply rook/base/ceph-for-ss/cluster.yaml:36]: Local storage provisioner creates PV with whole disk capacity, even if storage request is specified. However this settings is confusing , so we'll add a comment.

File: rook/base/ceph-for-cs/cephblockpool.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+kind: CephBlockPool
+metadata:
+  name: cs-block-pool
+  namespace: ceph-for-cs
+  annotations:
+    argocd.argoproj.io/sync-wave: "2"
+spec:
+  failureDomain: zone
+  replicated:
+    size: 3
```
[the remark this answers is not in the document]
Comment (ysksuzuki) [reply rook/base/ceph-for-cs/cephblockpool.yaml:11]: We'll work on it in the following task.
https://app.zenhub.com/workspaces/neco-5ca28d055128c041a68c0a8f/issues/cybozu-go/neco/755

[the remark this answers is not in the document]
Comment (ymmt2005) [reply rook/base/ceph-for-cs/cephblockpool.yaml:11]: I don't think erasure coding is preferable for our use cases.
Simple replication makes things simple and fast.

**Pull request merged**